src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,251 @@
+package edu.berkeley.ce.rockslicing
+
+import scala.annotation.tailrec
+
+/**
+  * Selects joints from the input joint set that will be used to seed the RDD in order to maintain
+  * approximately balanced loads among processors.
+  */
+object SeedJointSelector {
+  val THRESHOLD = 0.50
+
+  /**
+    * Searches though sequence of joint sets to find best seed joints
+    *
+    * @param jointSets Seq of Seq's of Joints representing input joint sets
+    * @param inputVolume Block representing input rock volume
+    * @param numProcessors Number of processors to be used in the analysis
+    * @param remainingJoints Seq of Seq of joints that were not selected as seed joints
+    * @return Returns a tuple containing two Seq's of Joints. The first Seq contains the seed joints that
+    *         best divide the input volume. The second Seq contains the joints that remain from the input
+    *         joint set.
+    */
+  @tailrec
+  def searchJointSets(jointSets: Seq[Seq[Joint]], inputVolume: Block, numProcessors: Int,
+                      remainingJoints: Seq[Seq[Joint]] = Seq.empty[Seq[Joint]]): (Seq[Joint], Seq[Joint]) = {
+    val allJoints = if (remainingJoints.isEmpty) {
+      jointSets
+    } else {
+      remainingJoints
+    }
+
+    if (jointSets.isEmpty) {
+      throw new IllegalStateException("ERROR: Unable to find satisfactory seed joints")
+    }
+    if (jointSets.head(0).shape.isEmpty) {
+      // Check that joint set is persistent
```

Comment:
If this comment applies to the `if` on the previous line, I think it should come on the line before the `if`. I was kind of confused if it was about the conditional or the call to `findSeedJoints`.